nodeadm/api/v1alpha1/nodeconfig_types.go

```diff
@@ -29,19 +30,31 @@ type NodeConfigList struct {
 }
 
 type NodeConfigSpec struct {
-	Cluster      ClusterDetails  `json:"cluster,omitempty"`
-	FeatureGates map[string]bool `json:"featureGates,omitempty"`
+	Cluster ClusterDetails `json:"cluster,omitempty"`
+	Kubelet KubeletOptions `json:"kubelet,omitempty"`
 }
 
 type ClusterDetails struct {
-	// +kubebuilder:validation:Required
-	Name string `json:"name,omitempty"`
-	// +kubebuilder:validation:Required
-	APIServerEndpoint string `json:"apiServerEndpoint,omitempty"`
-	// +kubebuilder:validation:Required
+	Name                 string `json:"name,omitempty"`
+	APIServerEndpoint    string `json:"apiServerEndpoint,omitempty"`
 	CertificateAuthority []byte `json:"certificateAuthority,omitempty"`
-	// +kubebuilder:validation:Required
-	CIDR          string `json:"cidr,omitempty"`
-	EnableOutpost *bool  `json:"enableOutpost,omitempty"`
-	ID            string `json:"id,omitempty"`
+	CIDR                 string `json:"cidr,omitempty"`
+	EnableOutpost        *bool  `json:"enableOutpost,omitempty"`
+	ID                   string `json:"id,omitempty"`
+}
+
+type KubeletOptions struct {
+	// InlineConfig is a raw document of a kubelet config that can be provided
+	// by the user to override default generated configurations
+	InlineConfig string `json:"inline,omitempty"`
+	// Flags is a string map that takes command-line kubelet arguments without
+	// leading dashes. These arguments override any of generated defaults
+	// https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/
+	Flags map[string]string `json:"flags,omitempty"`
+	// Labels is a map of labels to apply to the node when the kubelet
+	// registers itself
+	Labels map[string]string `json:"labels,omitempty"`
+	// Labels is a map of labels to apply to the node when the kubelet
+	// registers itself
+	Taints []v1.Taint `json:"taints,omitempty"`
```

Comment:
Is this a premature addition of the `labels` and `taints` fields at the top level? they are technically configurable through `inline` config or `flags`, and i dont think we need to do any special validation with them.

Many users might expect `labels` and `taints` coming from the previous bootstrap.sh, but there's not a need to uphold that image

Reply:
Managed nodegroups has passed these with the `--kubelet-extra-args` flag, so I think it's fine to let orchestrators use the generic kubelet options directly.

Reply:
ill pull back on these for now, since an operator can still fully configure kubelet without them and the contracts for these via `--node-labels`, `--register-with-taints` isn't going to break anytime soon